src/Modules/Data/Commands.Data/AddPowerBIRows.cs

```diff
@@ -0,0 +1,92 @@
+﻿/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+using System;
+using System.Management.Automation;
+using Microsoft.PowerBI.Common.Api.Datasets;
+using Microsoft.PowerBI.Common.Api.Workspaces;
+using Microsoft.PowerBI.Common.Client;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.PowerBI.Commands.Data
+{
+    [Cmdlet(CmdletVerb, CmdletName, DefaultParameterSetName = DatasetParameterSetName)]
+    [OutputType(typeof(Dataset))]
+    public class AddPowerBIRows : PowerBIClientCmdlet
+    {
+        public const string CmdletVerb = VerbsCommon.Add;
+        public const string CmdletName = "PowerBIRows";
+
+        #region ParameterSets
+        private const string DatasetIdParameterSetName = "DatasetId";
+        private const string DatasetParameterSetName = "Dataset";
+        #endregion
+
+        #region Parameters
+        [Parameter(Mandatory = true, ParameterSetName = DatasetIdParameterSetName)]
+        public Guid DatasetId { get; set; }
+
+        [Parameter(Mandatory = true, ValueFromPipeline = true, ParameterSetName = DatasetParameterSetName)]
+        public Dataset Dataset { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = DatasetIdParameterSetName)]
+        [Parameter(Mandatory = true, ParameterSetName = DatasetParameterSetName)]
+        public string TableName { get; set; }
+
+        [Parameter(Mandatory =true)]
+        public JArray Rows { get; set; }
```

Comment:
I'm wondering if there is a way we could encapsulate this because we are likely to move away from Newtonsoft at some point in the future to reduce dependency issues

Reply:
Done